packages/cli/src/lib/snapshot/snapshotReporter.ts

```diff
@@ -57,6 +58,39 @@ export class SnapshotReporter {
     );
   }
 
+  public getReportStatus(): SnapshotReportStatus {
+    if (this.isSuccessReport()) {
+      return SnapshotReportStatus.SUCCESS;
+    }
+    if (this.isVaultEntriesMissingReport()) {
+      return SnapshotReportStatus.MISSING_VAULT_ENTRIES;
+    }
+    return SnapshotReportStatus.ERROR;
+  }
+
+  private isVaultEntriesMissingReport(): boolean {
+    for (const resourceTypeEntry of Object.values(
+      this.report.resourceOperationResults
+    )) {
+      for (const errors of Object.values(resourceTypeEntry)) {
+        for (const err of errors) {
+          if (!SnapshotReporter.isVaultEntryMessage(err)) {
+            return false;
+          }
+        }
+      }
+    }
+    return true;
+  }
```

Comment:
Ain't a fan, but I think it's the fastest.
The fastest way to know if all errors are missing vault entries errors is to check them all and return false as soon as we find an error that ain't a missing vault entry.